examples/contract.golden

```diff
@@ -1,5 +1,5 @@
-#<closure> : (Syntax → Syntax)
-#<closure> : (Syntax → (Syntax → (Syntax → (Macro Syntax))))
-#<closure> : (Syntax → (Syntax → (Syntax → (Macro Syntax))))
-#<closure> : ∀(α : *). (α → α)
+#<contract> : (Syntax → Syntax)
+#<arg> : (Syntax → (Syntax → (Syntax → (Macro Syntax))))
```

Comment:
looks like things defined with `defun` are still capturing the argument not the function name

Reply:
- [ ] : TODO fix defun

Reply:
I think these do not make it to the evaluator. They are wired into the expander in the kernel. This means they are defined as `decl`s here:
```
evalMod (Expanded em _) = execStateT (traverseOf_ (moduleBody . each) evalDecl em) mempty
  where
    evalDecl :: CompleteDecl -> StateT (Seq EvalResult) Expand ()
    evalDecl (CompleteDecl d) =
...
```

Reply:
And the `Decls` are not represented by `Core` so the evaluator typically only sees the body of the `Decl` which makes sense. This is a good use case for a passthrough Kont. But I'll save that for a different PR.